Source/JavaScriptCore/runtime/MathObject.cpp

```diff
@@ -190,13 +190,12 @@ JSC_DEFINE_HOST_FUNCTION(mathProtoFuncHypot, (JSGlobalObject* globalObject, Call
     auto scope = DECLARE_THROW_SCOPE(vm);
 
     unsigned argsCount = callFrame->argumentCount();
-    Vector<double, 8> args;
-    args.reserveInitialCapacity(argsCount);
-    for (unsigned i = 0; i < argsCount; ++i) {
+    Vector<double, 8> args(argsCount, [&](size_t i) -> std::optional<double> {
         double argument = callFrame->uncheckedArgument(i).toNumber(globalObject);
-        RETURN_IF_EXCEPTION(scope, { });
-        args.unsafeAppendWithoutCapacityCheck(argument);
-    }
+        RETURN_IF_EXCEPTION(scope, std::nullopt);
```

Comment:
The old code used to bail out if an exception happened without doing further work. This code instead continues to evaluate future arguments although it \then does return. We should not change that semantic.

Reply:
The new code bails out when an exception happens too. I do not think I changed behavior. This constructor is not a "compactMap"-like constructor:
```
            for (size_t i = 0; i < size; ++i) {
                if (auto item = valueGenerator(i))
                    unsafeAppendWithoutCapacityCheck(WTFMove(*item));
                else
                    return;
            }
```

Reply:
Maybe we should clarify this in some way? Maybe we could add a template parameter to specify whether to skip or abort when getting std::nullopt?